### ISSUE TITLE
WCAG issue - Color contrast of checkboxes in theming is insufficient on certain thumbnails

### ISSUE THREAD

**Issue by Rainribbon** [Kaltura-EDU/issues-tracking#20]
Submitted to Kaltura in Case #00549175:

With theming enabled in KAF, the checkboxes in My Media have been moved to overlap the thumbnail images. Depending on the colors included in the thumbnail, the checkboxes can be very difficult to see.

Taking a screenshot of the page and attempting to sample the colors on a thumbnail where the checkbox is barely visible, the checkbox comes in as hex #949798. On a light gray or white background, this color fails the minimum color contrast requirements outlined in [WCAG Success Criterion 1.4.3: Contrast (Minimum) (Level AA)](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum)

![Image](https://github.com/user-attachments/assets/29924c6a-8e41-43d7-a3dc-f836aa1ece79)

Recommendation: Move the checkboxes back outside the thumbnails so they are clearly visible to all users.

**Comment by Rainribbon** [comment]
Kaltura is tracking this issue under ADA-2830: WCAG issue – Checkbox color contrast insufficient when overlapping thumbnail in themed KAF

**Comment by Rainribbon** [comment]
I sent the following message in response to my related support ticket today:

I wanted to follow up on this issue because I discovered an additional keyboard accessibility problem with the checkboxes being on top of the thumbnails - because they overlap, it's easy to get confused as to which item is in currently in focus (the checkbox or the thumbnail). I encountered this issue the first time I tried to make a recording for Case #00549538 - I had accidentally tabbed onto the thumbnail without realizing it, so pressing spacebar to try to select the checkbox skipped me down on the page. This led me to believe at first that the checkboxes themselves were inaccessible. A few seconds later, I was able to select the checkboxes correctly (presumably, because I tabbed down to get a different one in focus), so I re-recorded the video for Case #00549538.

Steps to reproduce:
1. In My Media with Theming enabled, have or add enough media entries that you must scroll down in your browser window to see some of them on the screen.
2. Tab down to a checkbox for an entry. Press Tab again.
3. Acting as a user who didn't realize they just tabbed away from the desired checkbox because the checkbox and thumbnail are overlapping each other, press spacebar in an attempt to select the checkbox.
4. Note that your screen scrolls down in a jarring way. If you press Tab again, the screen scrolls back up and you find the checkbox was not selected. This makes it seem like the checkbox is completely inaccessible.

Given this issue as well, I think it's even more important to revert the visual merger of the checkbox and thumbnail so it's clear to the user which item is in focus.

**Comment by Rainribbon** [comment]
Kaltura stated that they created two tickets related to this issue:

- ADA-2830 – WCAG issue: Checkbox color contrast insufficient when overlapping thumbnail in themed KAF
- ADA-2885 – Keyboard accessibility issue: Overlapping checkboxes and thumbnails in My Media

They have now shared the following:

> We received some feedback from our Tier 3 ADA team regarding the issue in ADA-2885: Keyboard accessibility issue due to overlapping checkboxes and thumbnails in My Media.
> 
> After reviewing the behavior, this appears to be the intended functionality rather than a product defect. Keyboard users can tab to the checkbox, use the Spacebar to select or deselect it, and then continue tabbing to the thumbnail. When the thumbnail receives focus, pressing Enter opens the media entry.
> 
> The checkbox remains keyboard accessible and can be operated as expected using the Spacebar when it has focus.
> 
> Based on the current design and interaction model, we do not consider this a bug. Please review and provide any feedback when you have a moment.

I sent the following response:

I'm afraid having the checkbox and thumbnail overlap is still an accessibility failure, and I consider it among the primary blockers to implementation of Theming. As noted in the [Web Content Accessibility Guidelines (WCAG) 2.1](https://www.w3.org/TR/WCAG21/#dfn-targets): "If two or more targets are overlapping, the overlapping area should not be included in the measurement of the target size, except when the overlapping targets perform the same action or open the same page."

The behavior of not being able to get focus or click on the correct element easily negatively impacts users in multiple disability categories:
1. Visual (low vision) - As I demonstrated in my video, even a person like me without significant visual impairment can have difficulty telling which element they are focused on (checkbox vs. thumbnail) and thus attempt to interact with the wrong element. This will likely pose additional challenges for users with partially obscured vision.
2. Visual (color) - With some thumbnails, color contrast of the checkbox is insufficient.
3. Mobility/Flexibility/Body Structure - Users who have difficulty with motor control cannot easily click on the checkbox while avoiding clicking on the thumbnail image, thus being directed to the wrong page. ([Failure of Target Size (Minimum) (Level AA)](https://www.w3.org/WAI/WCAG22/Understanding/target-size-minimum.html))
4. Cognitive - Users with cognitive disabilities will also be confused by the inability to select the desired target with the targets overlapping.
